packages/client/src/devices/devices.ts

```diff
@@ -151,10 +156,17 @@ export const getAudioOutputDevices = () => {
     concatMap(() => getDevices(getAudioBrowserPermission(), 'audiooutput')),
     shareReplay(1),
   );
-};
+});
 
 const getStream = async (constraints: MediaStreamConstraints) => {
-  return await navigator.mediaDevices.getUserMedia(constraints);
+  const stream = await navigator.mediaDevices.getUserMedia(constraints);
+  // On Firefox, there's no reliable way to listen to device permission changes,
+  // so it's difficult to update the device list once camera or microphone access
+  // is allowed by the user. However, if we were able to get user media, we can be
+  // sure permission was granted, so we fake the devicechange event to force device
+  // lists to update.
+  navigator.mediaDevices.dispatchEvent(new Event('devicechange'));
```

Comment:
Shall we limit this dispatch to Firefox only?

Reply:
Done in [8ba1e8c](https://github.com/GetStream/stream-video-js/pull/1541/commits/8ba1e8ce11a2b6a00330f2c6dc257cd6985c7576). I also updated the comment, since it was misleading about the reasons to force this event.